### PR TITLE
Limit the number of comments showed.

### PR DESCRIPTION
**Pull request by Giost**
Allow the user to choose how many comments to view.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -40,12 +40,27 @@ private String convertListToJson(List<Comment> comments) {
...
   }
 
+  /**
+   * Return the value of the limit parameter if valid, 
+   * otherwise the default value which is 5
+   */
+  private int getLimit(HttpServletRequest request) {
+    int limit;
+    try {
+      limit = Integer.parseInt(getParameter(request, "limit", ""));
```
Comment (dola) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: If you want you could create a `getIntParameter` method that returns an `int` and the default value (which would also be an int) if it is not present or fails to parse.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -81,6 +81,15 @@ <h3><a href="https://github.com/Giost">GitHub</a></h3>
               <h2>Comments</h2>
           </div>
           <div class="description">
+              <div class="comment-view">
+                <label for="comment-limit">Select the number of comments to show: </label>
```
Comment (dola) [inline portfolio/src/main/webapp/index.html:85]: good call on the label!

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -132,12 +132,26 @@ function randomSlide(randomButton) {
...
 }
 
+/**
+ * Returns the value selected for the limit of comments to show.
+ */
+function getCommentsLimit() {
+  let limit = parseInt(document.getElementById("comment-limit").value);
+  if (isNaN(limit)) {
+    // if the limit is not valid, set it to the default value of 5
+    limit = 5;
```
Comment (dola) [inline portfolio/src/main/webapp/script.js]: I recommend extracting any constant values like this into a well-named constant. e.g. `DEFAULT_COMMENT_LIMIT`.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -40,12 +40,27 @@ private String convertListToJson(List<Comment> comments) {
...
+    } catch (Exception e) {
+      // if there is an error, returns the default value (5)
+      limit = 5;
```
Comment (dola) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: As below, extract this into a constant.

**Review by dola — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (Giost) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: It makes sense.

**Review by zlorf — APPROVED** [review]

**Pull request merged**